pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -81,14 +68,13 @@ type InstanceListOpts struct {
 	Name string `q:"name"`
 }
 
-func NewInstanceService(cfg *CloudConfig) (*InstanceService, error) {
-	authUrl := gophercloud.NormalizeURL(cfg.AuthURL)
+func NewInstanceService(cfg *clientconfig.Cloud) (*InstanceService, error) {
 	opts := &gophercloud.AuthOptions{
```

Comment:
Could you add all the struct fields? https://github.com/flaper87/terraform-provider-openstack/blob/master/openstack/config.go#L103-L115

That way we won't be leaving functionality out :smile:

Reply:
@flaper87 clientconfig.Cloud has all the fields of that struct and then some. https://github.com/gophercloud/utils/blob/259b9dbae8cbb5aa5b61d08555d98a7c522e736e/openstack/clientconfig/results.go#L18

is it worth duplicating?

Reply:
We clarified this on IRC. Resolving this convo :+1: